lib/net.js

```diff
@@ -322,6 +322,11 @@ Socket.prototype._onTimeout = function() {
 
 
 Socket.prototype.setNoDelay = function(enable) {
+  if (!this._handle) {
+    this.once('connect', this.setNoDelay.bind(this, enable));
```

Comment:
I'm not sure if it's a worthy optimization but `this.setNoDelay()` is the same as `this.setNoDelay(true)`.  That is, you could avoid the .bind call in what is presumably the common case:

```javascript
this.once('connect', enable ? this.setNoDelay : this.setNoDelay.bind(this, false));
```

Your call.